java/ql/src/experimental/Security/CWE/CWE-601/SpringUrlRedirect.ql

```diff
@@ -0,0 +1,42 @@
+/**
+ * @name Spring url redirection from remote source
+ * @description Spring url redirection based on unvalidated user-input
+ *              may cause redirection to malicious web sites.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/spring-unvalidated-url-redirection
+ * @tags security
+ *       external/cwe-601
+ */
+
+import java
+import SpringUrlRedirect
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class SpringUrlRedirectFlowConfig extends TaintTracking::Configuration {
+  SpringUrlRedirectFlowConfig() { this = "SpringUrlRedirectFlowConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof SpringUrlRedirectSink }
+
+  override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
+    guard instanceof StartsWithSanitizer
+  }
+
+  override predicate isSanitizer(DataFlow::Node node) {
+    // Exclude the case where the left side of the concatenated string is not `redirect:`.
+    // E.g: `String url = "/path?token=" + request.getParameter("token");`
+    exists(AddExpr ae |
+      ae.getRightOperand() = node.asExpr() and
+      not ae instanceof RedirectBuilderExpr
```

Comment:
it's great. I will add and verify immediately.

Reply:
It would be nice to also account for prefixes in format strings such as in https://lgtm.com/projects/g/Yuri-Krukovski/my-sample/snapshot/767b6c59de570cfbfeeaf81188488bd8d2958b56/files/ngo-admin/src/main/java/ngo/skarb/ui/controller/story/SuccessStoryStatusChangeController.java?sort=name&dir=ASC&mode=heatmap#L76

So that if the tainted var is not in the first argument position or if the string format does not start with a placeholder, we can consider it as a prefixed string

Reply:
Ok